### PR TITLE
Docker support to run the project

### PR DESCRIPTION
**Pull request by sub2u**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/332)

### CODE REVIEW

**Comment by sub2u** [comment]
Referring to https://github.com/shakacode/react-webpack-rails-tutorial/issues/331

**Comment by justin808** [comment]
@dylangrafmyre Can you please review this one.

**Comment by dylangrafmyre** [comment]
@sub2u Could you please link to your `sub2u/ruby-2.3.1-node-6.8.0` Dockerfile. I do not see it on Docker Hub. Is it in a Github repo?

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/332#-:-KUsWRNlfYMSx1mOi3Sb:b-tp2mgh)*

**Comment by sub2u** [comment]
@dylangrafmyre You can find the docker in docker hub
https://hub.docker.com/r/sub2u/ruby-2.3.1-node-6.8.0/

**Comment by dylangrafmyre** [comment]
@sub2u there is no `Dockerfile` tab under your referenced docker hub repo.


---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/332#-:-KUttW28o-QyLVFpsK4l:bqzbj8q)*

**Comment by sub2u** [comment]
@dylangrafmyre Docker file is in github repo please review in https://github.com/sub2u/ruby-2.3.1-node-6.8.0

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM sub2u/ruby-2.3.1-node-6.8.0
+MAINTAINER sub2u
+
+COPY ./ app/
```
Comment (alexfedoseev) [inline Dockerfile]: Wouldn't it be more convenient to mount directory instead of coping over all the things inside container?

**Comment by alexfedoseev** [comment]
@sub2u IMO you have too many layers in https://github.com/sub2u/ruby-2.3.1-node-6.8.0

Your stuff will be outdated if you will change something below `RUN apt-get update -qq ...` as this layer is cached by the docker and will be reused.

File: Dockerfile (hunk 1 above)
Comment (sub2u) [inline Dockerfile]: Sure, we can use ADD, i will update it

**Comment by sub2u** [comment]
@alexfedoseev we can use force update, in `RUN apt-get update -qq` but j am not sure if will effect the build performance it always try to get updates instead of fetching from caching layer.

File: Dockerfile (hunk 1 above)
Comment (alexfedoseev) [inline Dockerfile]: I was meaning not `ADD`, but [volumes](https://docs.docker.com/engine/tutorials/dockervolumes/#/mount-a-host-directory-as-a-data-volume). Copying process takes time and in development copy files on change is kinda pricy. If volume is used, then it reads the files directly from the mounted volume, so it should be faster.

**Comment by alexfedoseev** [comment]
> we can use force update, in RUN apt-get update -qq but j am not sure if will effect the build performance it always try to get updates instead of fetching from caching layer.

If it acts like this then I believe there's something wrong is going on w/ this setup as unchanged layer must be taken from the cache: https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/minimize-the-number-of-layers

**Comment by dylangrafmyre** [comment]
@sub2u Thank you for taking the time to work on this. For this project I think it would be best to have a project Dockerfile  that we maintained off of the the Ruby 2.3.1-alpine base image or similar. I  do not think it is best to rely on a maintained image by contributors to this project. 

@alexfedoseev brings up very good points about image size and minimizing image layers. 

I would be more than happy to work on getting the right Dockerfile config and build in place with you that meets the needs of this tutorial.

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/332#-:-KUuJbtl_TcgA6H3x0x4:b-fisakc)*

**Comment by sub2u** [comment]
Sure @dylangrafmyre i will update the docker file

**Comment by sub2u** [comment]
Yes agreed Image size should be less,


It contain All Ruby nodejs, and other libraries required for project it is 366 MB size.
I will prepare the docker file with volume to link from external volumes.

**Comment by justin808** [comment]
@sub2u Any update?

**Comment by justin808** [comment]
@sub2u Anything I can do to help?

**Pull request closed without merging**